### PR TITLE
Implement mechanism for DNS federation

### PR DESCRIPTION
**Pull request by bcdonadio**
Documentation with examples included. Should implement the internal parts of the role needed for federation, because our open/closed model sucks and needs a second, if not third, external part to work.

### CODE REVIEW

File: roles/kubernetes-apps/ansible/templates/coredns-config.yml.j2 [hunk 1]
```diff
@@ -29,13 +29,38 @@ data:
...
+        loop
+        reload
+        loadbalance
+    }
+{%   for block in coredns_federation_zones %}
+{% if clusterIP != block['nameserver'] %}
+    {{ block['zone'] }}.:53 {
+        import fedplugins
+        rewrite stop name suffix .{{ block['zone'] }}. .{{ block['suffix'] | default(dns_domain) }}. answer auto
+        forward . {{ block['nameserver'] | join(' ') }}
```
Comment (guilhermeoki) [inline roles/kubernetes-apps/ansible/templates/coredns-config.yml.j2]: Why did you add this join here?

Comment (bcdonadio) [reply roles/kubernetes-apps/ansible/templates/coredns-config.yml.j2]: You can have more than one resolving upstream endpoint. Not that we would use, but it maintains the same general structure as the implementation of external zones a few lines above.

**Review by guilhermeoki — APPROVED** [review]
LGTM!

File: roles/kubernetes-apps/ansible/templates/coredns-config.yml.j2 (hunk 1 above)
Comment (bcdonadio) [reply roles/kubernetes-apps/ansible/templates/coredns-config.yml.j2]: Fixed on e90a0c31
